commands/resource_commands/deploy

```diff
@@ -0,0 +1,247 @@
+#!/bin/bash
+set -e
+. "$GRPL_WORKDIR/utils/help_menus"
+. "$GRPL_WORKDIR/utils/checks"
+
+cli_help() {
+  grpl_resource_deploy_cli_help
+  exit 0
+}
+
+[ "$1" = "help" ] || [ "$1" = "h" ] && cli_help
+
+install_prerequisite
+
+cli_log "resource deploy BEGIN"
+status_log $TYPE_DEFAULT "resource deploy is in progress"
+NS=grpl-system
+DB_FILE_NS=grpl-dbfile
+DB_NS=grpl-db
+awsregistry="p7h7z5g3"
+
+
+
+
+# ----------------------- functions code ---------------------------
+
+transform_model_input_to_yaml() {
+
+    check_and_install_jq
+    check_and_install_yq
+
+    local Models=$1
+    # Convert the JSON-like string to valid JSON by replacing single quotes with double quotes
+    json_string="${Models//\'/\"}"
+
+    # Initialize an array to store the models
+    local models=()
+    models_name=()
+    # Split the input string by pipe symbol (|)
+    IFS='|' read -ra models_data <<< "$json_string"
+
+
+    # Iterate over each model data
+    for model_data in "${models_data[@]}"; do
+        # Extract the model name and its properties
+        IFS=':' read -r model_name properties <<< "$model_data"
+        
+        # Parse the properties JSON and format it as YAML
+        yaml_properties=$(echo "$properties" | jq -r '. | {spec: {name: .name, base: .base, properties: (.properties | map_values(with_entries(select(.value != null))))}}' | yq eval -P)
```

Comment:
I would appreciate here as little customization as possible.
can't we just convert JSON to YAML?

Reply:
we can't convert json directly, because of spec property that exist in yaml and not in json